intellij-ext/modules/products/pycharm/src/main/kotlin/com/metalbear/mirrord/products/pycharm/PythonRunConfigurationExtension.kt

```diff
@@ -0,0 +1,49 @@
+package com.metalbear.mirrord.products.pycharm
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.configurations.RunnerSettings
+import com.intellij.execution.target.createEnvironmentRequest
+import com.intellij.execution.wsl.WslPath.Companion.getDistributionByWindowsUncPath
+import com.intellij.execution.wsl.target.WslTargetEnvironmentRequest
+import com.intellij.openapi.util.SystemInfo
+import com.jetbrains.python.run.AbstractPythonRunConfiguration
+import com.jetbrains.python.run.PythonRunConfigurationExtension
+import com.metalbear.mirrord.MirrordExecManager
+
+
+class PythonRunConfigurationExtension: PythonRunConfigurationExtension() {
+    override fun isApplicableFor(configuration: AbstractPythonRunConfiguration<*>): Boolean {
+        return true
+    }
+
+    override fun isEnabledFor(
+        applicableConfiguration: AbstractPythonRunConfiguration<*>,
+        runnerSettings: RunnerSettings?
+    ): Boolean {
+        return true
+    }
+
+
+    override fun patchCommandLine(
+        configuration: AbstractPythonRunConfiguration<*>,
+        runnerSettings: RunnerSettings?,
+        cmdLine: GeneralCommandLine,
+        runnerId: String
+    ) {
+        val wsl = when (val request = createEnvironmentRequest(configuration, configuration.project)) {
```

Comment:
Is this actually required in addition to the extendTargetEnvironment bit?

Reply:
Those are two different extensions - I suppose there are cases where each used